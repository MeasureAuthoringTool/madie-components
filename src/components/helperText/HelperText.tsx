import React from "react";
import tw, { styled } from "twin.macro";

export interface HelperTextProps extends React.HTMLAttributes<HTMLSpanElement> {
  text: string;
  isError?: boolean;
}

const defaultHelperText = tw.span`mt-2 text-sm text-gray-600 dark:text-gray-300`;
const errorHelperText = tw(defaultHelperText)`text-red dark:text-red-300`;

export function HelperTextComponent(props: HelperTextProps) {
  const { text, isError, ...args } = props;

  const StyledHelperTextSpan = !!isError ? errorHelperText : defaultHelperText;

  return <StyledHelperTextSpan {...args}>{text}</StyledHelperTextSpan>;
}
