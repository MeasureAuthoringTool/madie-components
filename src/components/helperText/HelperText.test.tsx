import * as React from "react";
import { render } from "@testing-library/react";
import HelperText from "./HelperText";

const testId = "helper-text-component-test";
const defaultText = "Help text goes here";
const errorText = "Error text goes here";

describe("Helper Text Component", () => {
  it("should render a default helper text", () => {
    const { getByTestId } = render(
      <HelperText data-testid={testId} text={defaultText} />
    );
    const helperText = getByTestId(testId);
    expect(helperText).toBeInTheDocument();
    expect(helperText).toHaveTextContent(defaultText);
    expect(helperText).toMatchSnapshot();
  });

  it("should render a error helper text", () => {
    const { getByTestId } = render(
      <HelperText data-testid={testId} text={errorText} isError={true} />
    );
    const helperText = getByTestId(testId);
    expect(helperText).toBeInTheDocument();
    expect(helperText).toHaveTextContent(errorText);
    expect(helperText).toMatchSnapshot();
  });
});
