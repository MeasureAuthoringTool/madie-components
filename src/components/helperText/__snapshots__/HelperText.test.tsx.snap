// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helper Text Component should render a default helper text 1`] = `
<span
  class="HelperText__defaultHelperText-sc-4k037i-0 iJGOzB"
  data-testid="helper-text-component-test"
>
  Help text goes here
</span>
`;

exports[`Helper Text Component should render a error helper text 1`] = `
<span
  class="HelperText__defaultHelperText-sc-4k037i-0 HelperText__errorHelperText-sc-4k037i-1 iJGOzB ewRCez"
  data-testid="helper-text-component-test"
>
  Error text goes here
</span>
`;
